### PR TITLE
EVA-3524: Catch and report HTTP errors

### PR DESCRIPTION
**Pull request by apriltuesday**
Tried to report partial results where possible while still having the check show up as failing. This makes the final report a bit confusing, but if e.g. one sequence was already found to be not INSDC before ENA failed, that would be shown.

Screenshot from the test - the idea trying to be conveyed here is that nothing amiss was found so far, but the check is incomplete and should be rerun.
image

### CODE REVIEW

File: bin/check_fasta_insdc.py [hunk 1]
```diff
@@ -107,23 +105,29 @@ def assess_fasta(input_fasta, analyses, assembly_in_metadata):
...
+                    logger.warning(f'Sequence with this MD5 is INSDC but not found in contig alias: {md5_digest}')
+            elif len(possible_assemblies) == 0:
+                possible_assemblies = containing_assemblies
+            else:
+                possible_assemblies &= containing_assemblies
+    except HTTPError as e:
+        # Server errors from either ENA refget or EVA contig alias will halt the check prematurely.
+        # Report the error but do not return from the method, so that incomplete results can be reported
+        # (i.e. any sequences found to be INSDC and any compatible assemblies so far)
+        results['error'] = str(e)
```
Comment (tcezard) [inline bin/check_fasta_insdc.py:130]:
```suggestion
        results['connection_error'] = str(e)
```

File: eva_sub_cli/jinja_templates/fasta_check.html [hunk 2]
```diff
@@ -1,8 +1,17 @@
 
 {% macro fasta_check_report(validation_results, file_name) -%}
     {% set results_for_fasta = validation_results['fasta_check'][file_name] %}
-    {% set expand_icon = "" %}
+
+    {% if results_for_fasta.get('error') %}
```
Comment (tcezard) [inline eva_sub_cli/jinja_templates/fasta_check.html:5]:
```suggestion
    {% if results_for_fasta.get('connection_error') %}
```

File: eva_sub_cli/jinja_templates/fasta_check.html [hunk 3]
```diff
@@ -1,8 +1,17 @@
...
+        <div class="report-section fail">
+            <strong>Warning:</strong> The following results may be incomplete due to problems with external services. Please try again later for
+            complete results.
+            <br><strong>Error message:</strong> {{ results_for_fasta.get('error') }}
```
Comment (tcezard) [inline eva_sub_cli/jinja_templates/fasta_check.html:9]:
```suggestion
            <br><strong>Error message:</strong> {{ results_for_fasta.get('connection_error') }}
```

**Review by tcezard — APPROVED** [review]
It looks good to me.

**Review by nitin-ebi — APPROVED** [review]

**Pull request merged**